src/Serilog.Sinks.RollingFile/Sinks/RollingFile/TemplatedPathRoller.cs

```diff
@@ -29,18 +29,42 @@ class TemplatedPathRoller
         const string OldStyleDateSpecifier = "{0}";
         const string DateSpecifier = "{Date}";
         const string DateFormat = "yyyyMMdd";
+        const string HourSpecifier = "{Hour}";
+        const string HourFormat = "yyyyMMddHH";
+        const string HalfHourSpecifier = "{HalfHour}";
+        const string HalfHourFormat = "yyyyMMddHHmm";
         const string DefaultSeparator = "-";
 
+        const string MatcherMarkSpecifier = "date";
+        const string MatcherMarkInc = "inc";
+
         readonly string _pathTemplate;
         readonly Regex _filenameMatcher;
+        readonly SpecifierTypeEnum _specifierType = SpecifierTypeEnum.None;
```

Comment:
One other option that would work really well:

```csharp
class Specifier
{
    public string Name { get; }
    public string Token { get; }
    public string Format { get; }
    public TimeSpan Interval { get; }

    Specifier(string name, string format, string interval)
    {
        Name = name;
        Token = "{" + Name + "}";
        Format = format;
        Interval = interval;
    }

    public static readonly Specifier Date = new Specifier("Date", "yyyyMMdd", TimeSpan.FromDays(1));
    public static readonly Specifier Hour = new Specifier("Hour", "yyyyMMddhh", TimeSpan.FromHours(1));
    public static readonly Specifier HalfHour = new Specifier("HalfHour", "yyyyMMddhhmm", TimeSpan.FromMinutes(30));
}
```

We could then have a field:

```csharp
readonly Specifier _specifier;
```

Then, after construction, we could avoid checking the current specifier and just use fields like `_specifier.Token`, `_specifier.Interval` and so-on.

What do you think?